### DISCUSSION TITLE
Can't log out when using remote_user_guard

### DISCUSSION THREAD

**Discussion by chrisx8** [firefly-iii/firefly-iii#5379] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I'm running Firefly III 5.6.6 (latest version) with remote_user_guard. `CUSTOM_LOGOUT_URI` is configured. The "Log Out" button doesn't show on the sidebar, and the "Logout" option under Options/Profile causes 500 error.

### Expected behaviour

Should be able to log out when using remote_user_guard with `CUSTOM_LOGOUT_URI` is configured.

### Debug information

Error message:
```
Call to undefined method FireflyIII\Support\Authentication\RemoteUserGuard::logout()
```

Debug information generated at 2021-12-10 17:09:48 America/New_York for Firefly III version **5.6.6**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.6.6 |
| Firefly III API | 1.5.5 |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | 948b24a3-dff1-4677-968c-d6d2bddd58f0 |
| Using docker? | false |
| Layout | v1 |
| System TZ | America/New_York |
| Browser TZ | America/New_York |
| App environment | production |
| App debug mode | true |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql |
| Current driver | mysql |
| DB version | 18 (exp. 18) |
| Login provider | eloquent  |
| Trusted proxies (.env) | 10.69.233.11 |

| User info | Value |
| --- | --- |
| Session start | 2021-12-01 00:00:00 |
| Session end | 2021-12-31 23:59:59 |
| Session first | 2021-06-04 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.93 Safari/537.36 |

### Steps to reproduce

1. set up remote_user_guard and `CUSTOM_LOGOUT_URI`
2. Log into Firefly III
3. "Log out" button doesn't exist in sidebar
4. Click "Logout" on Options/Profile page. Returs 500 error with 
```
Call to undefined method FireflyIII\Support\Authentication\RemoteUserGuard::logout()
```

### Additional info

- This bug appeared after the 5.6.0 upgrade.
- Log out works as expected before 5.6.0. Tested on 5.5.13 with the same config.

**Comment by JC5** [comment, marked as answer]
The correct environment variable is `CUSTOM_LOGOUT_URL`
